tinygrad/tensor.py

```diff
@@ -1,34 +1,67 @@
 # inspired by https://github.com/karpathy/micrograd/blob/master/micrograd/engine.py
 from __future__ import annotations
-import math, functools, itertools, operator, time
+import time
+from functools import partialmethod, reduce
+from itertools import accumulate, filterfalse
+import operator
 import numpy as np
-from typing import List, Tuple, Callable, Optional, ClassVar, Type, Union, Sequence
-from tinygrad.helpers import prod, argfix, make_pair, getenv, IMAGE, DEBUG, flatten, DType, dtypes, ImageDType
+from typing import List, Tuple, Callable, Optional, ClassVar, Type, Union, Sequence, cast
+from tinygrad.helpers import argfix, make_pair, getenv, IMAGE, DEBUG, flatten, DType, dtypes
+from math import ceil, pi, prod, sqrt
 from tinygrad.lazy import Device, LazyBuffer
 from tinygrad.ops import LoadOps
 
 # An instantiation of the Function is the Context
 class Function:
```

Comment:
Hmm, I don't like this split into two classes. This is a lot a copy and paste. And it should support N arguments for CUSTOM functions. How much speed is this gaining?